sdk/core/AzureCore/Source/Pipeline/PipelineStage.swift

```diff
@@ -27,11 +27,11 @@
 import Foundation
 
 public typealias ResultHandler<TSuccess, TError: Error> = (Result<TSuccess, TError>, HTTPResponse?) -> Void
-public typealias HTTPResultHandler<T> = ResultHandler<T, Error>
-public typealias PipelineStageResultHandler = ResultHandler<PipelineResponse, PipelineError>
+public typealias HTTPResultHandler<T> = ResultHandler<T, AzureError>
+public typealias PipelineStageResultHandler = ResultHandler<PipelineResponse, AzureError>
 public typealias OnRequestCompletionHandler = (PipelineRequest, Error?) -> Void
 public typealias OnResponseCompletionHandler = (PipelineResponse, Error?) -> Void
-public typealias OnErrorCompletionHandler = (PipelineError, Bool) -> Void
+public typealias OnErrorCompletionHandler = (AzureError, Bool) -> Void
```

Comment:
Does it make sense any more to have an `OnErrorCompletionHandler` when our onRequest / onResponse handlers both take an optional Error object?

Reply:
So.... arguably yes or no? Maybe? This method is called if either `on(request:...)` or `on(response:...)` returns an error as a way to hopefully handle the error and avoid a failure scenario. So you could handle that logic in those respective methods, and if the logic is different that would certainly make sense. Currently, logging policy is the only policy we have that actually uses this method, so perhaps it would make sense to do away with it.

Reply:
Filed #357 to look into this another time